pkg/clean/types.go

```diff
@@ -41,4 +41,5 @@ var (
 	GetNSXClientFailed       = Status{Code: 2, Message: "failed to get nsx client"}
 	InitCleanupServiceFailed = Status{Code: 3, Message: "failed to initialize cleanup service"}
 	CleanupResourceFailed    = Status{Code: 4, Message: "failed to clean up"}
+	TimeoutFailed            = Status{Code: 5, Message: "failed because of timeout"}
```

Comment:
Slightly related - why not have `Status` implement the `error` interface? For consumers that want to use `error` handling, they can do so easily, while your `main.go` package can still interpret the error as a `Status`. e.g.

```
func Clean() error  {
    return TimeoutFailed
}
```

in `main.go`, it could then do something like:

```
err := clean.Clean()

var statusErr Status
if errors.As(err, &statusErr) {
    return statusErr.Code, Message, // etc.
}

// Otherwise, it's some generic standard processing error
```

Consumer's can then cast this way to make localizable errors too if they need to and it reduces the need for extra return values (which I would argue are confusing since the error responses are essentially duplicated).

Reply:
@seanpang-vmware What's your opinion？

Reply:
I'm OK with it.

Reply:
@ihgann pls help review https://github.com/vmware-tanzu/nsx-operator/pull/497, and I'll rebase it to continue modifying `context`